middleware.ts

```diff
@@ -0,0 +1,98 @@
+import { NextResponse } from 'next/server'
+import type { NextRequest } from 'next/server'
+import { getToken } from 'next-auth/jwt'
+
+// Define public routes that don't require authentication
+const publicRoutes = [
+  '/',
+  '/auth/signin',
+  '/auth/unauthorized',
+  '/auth/error',
+  '/api/auth',
+  '/faqs',
+  '/maintenance',
+  '/logo.svg',  // Explicitly allow logo access
+  '/favicon.ico',
+  '/manifest.json'
+]
+
+// Helper function to check if a route is public
+const isPublicRoute = (path: string) => {
```

Comment:
I am very lost in this page, why do we have 2 functions. If middleware is not run on these paths, why are you checking here?